app/models/host/managed.rb

```diff
@@ -537,11 +537,29 @@ def jumpstart?
     operatingsystem.family == "Solaris" and architecture.name =~/Sparc/i rescue false
   end
 
+  HOSTGROUP_INHERITED_ATTRIBUTES = %w{puppet_proxy_id puppet_ca_proxy_id environment_id compute_profile_id}
+
+  def apply_inherited_attributes(attributes)
```

Comment:
this method definitely deserves test coverage for both new and persisted records